### PR TITLE
feat(supercharge): add super props for supercharged tokens

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

Add new super props to track supercharged token and value.

### Other changes

N/A

### Tested

Updated unit tests, and manually tested

### How others should test

N/A

### Related issues

- Fixes RET-332

### Backwards compatibility

Yes

### CODE REVIEW

File: src/analytics/selectors.test.ts [hunk 1]
```diff
@@ -9,6 +9,25 @@ describe('getCurrentUserTraits', () => {
       account: { defaultCountryCode: '+33', pincodeType: PincodeType.CustomPin },
       goldToken: { balance: '1.01' },
       stableToken: { balances: { [Currency.Dollar]: '2.02', [Currency.Euro]: '3.03' } },
+      app: {
+        superchargeTokens: [
```
Comment (dievazqu) [inline src/analytics/selectors.test.ts:13]: We are deprecating this value, we should use `app.superchargeTokenConfigByToken` (see the changes at https://github.com/valora-inc/wallet/pull/2689)

File: src/analytics/selectors.ts [hunk 2]
```diff
@@ -51,6 +54,20 @@ export const getCurrentUserTraits = createSelector(
       }
     }
 
+    const { superchargingTokenConfig, hasMaxBalance } = balanceInfoForSupercharge
```
Comment (dievazqu) [inline src/analytics/selectors.ts:57]: With the new new changes you can get a  `superchargeUsdBalance` from the `superchargeInfoSelector`,

File: src/analytics/selectors.test.ts [hunk 3]
```diff
@@ -21,17 +40,19 @@ describe('getCurrentUserTraits', () => {
...
             symbol: 'cEUR',
             decimals: 18,
             imageUrl: '',
-            usdPrice: '1.2',
-            balance: '20',
+            usdPrice: '1.2345',
+            balance: '21',
             priceFetchedAt: Date.now(),
             isCoreToken: true,
+            isSupercharged: true,
```
Comment (dievazqu) [inline src/analytics/selectors.test.ts:55]: This `isSupercharged ` flag is not needed anymore with the new remote config

File: src/analytics/selectors.ts (hunk 2 above)
Comment (kathaypacific) [reply src/analytics/selectors.ts]: oh yeah my branch was behind your one. think it's up to date now!

**Comment by ValoraQA** [comment]
Hey @MuckT since is mentioned in how others should test section, we are skipping this PR

Cc: @kathaypacific 

Thanks..!

**Pull request merged**